### PR TITLE
RELATED: RAIL-3428 - Initial impl for use command & prereqs for it

### PR DESCRIPTION
**Pull request by lupko**
- only does extended input validation
- delivers fixes & refactorings to enable code reuse between actions

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: tools/plugin-toolkit/src/_base/credentials.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    credentials: BackendCredentials,
+): void {
+    const { username, password, token } = credentials;
+
+    if (backend === "bear") {
+        if (isEmpty(username)) {
+            throw new InputValidationError(
+                "username",
+                "",
+                "Unable to determine username to use when logging into GoodData platform. Please make sure GDC_USERNAME env variable is set in your session or in the .env file",
```
Comment (no23reason) [inline tools/plugin-toolkit/src/_base/credentials.ts:53]: These messages should mention .env.secrets file right?

File: tools/plugin-toolkit/src/_base/env.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    const dotEnvContent: Record<string, string> = readDotEnv();
+    const { GDC_USERNAME, GDC_PASSWORD, TIGER_API_TOKEN } = process.env;
+
+    if (backend === "bear" && GDC_USERNAME) {
+        logInfo(
+            "GDC_USERNAME env variable set for current session; going to use this for authentication and expecting GDC_PASSWORD is set for current session as well.",
+        );
+
+        dotEnvContent.GDC_USERNAME = GDC_USERNAME;
+        dotEnvContent.GDC_PASSWORD = GDC_PASSWORD ?? "";
```
Comment (no23reason) [inline tools/plugin-toolkit/src/_base/env.ts:35]: Nitpick: why this defaulting to empty string? Why is it there only for password?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5763/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7501/ : SUCCESS in 19m 08s

File: tools/plugin-toolkit/src/_base/credentials.ts (hunk 1 above)
Comment (lupko) [reply tools/plugin-toolkit/src/_base/credentials.ts]: fixed

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1894

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Pull request merged**